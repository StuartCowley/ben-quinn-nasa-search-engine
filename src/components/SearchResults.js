/* eslint-disable array-callback-return */
import React from "react";
import "../styles/searchResults.css";
import PropTypes from "prop-types";

function SearchResults({ results }) {
  if (!results.length) {
    return <p>Please enter a valid search item!</p>;
  }
  return (
    <>
      <p>Search results:</p>
      <div className="search-results">
        {results.map((item) => (
          <img
            key={results.indexOf(item)}
            data-testid="search-results__image"
            className="search-results__image"
            alt="space"
            src={item}
          />
        ))}
      </div>
    </>
  );
}

export default SearchResults;

SearchResults.defaultProps = {
  results: [],
};

SearchResults.propTypes = {
  results: PropTypes.arrayOf(PropTypes.string),
};
