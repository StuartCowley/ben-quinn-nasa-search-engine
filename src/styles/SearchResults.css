p {
    font-size: 1.2em;
    font-weight: bold;
}

.search-results__image {
    height: 20%;
    width: 20%;
    align-self: flex-start;
    margin-left: 50px;
    margin-top: 10px;
    border: 3px solid #FC3D21;
}