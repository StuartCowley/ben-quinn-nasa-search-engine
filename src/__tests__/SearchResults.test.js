import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<SearchResults />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders visible image", () => {
    const results = [""];

    const { getByTestId } = render(<SearchResults results={results} />);
    expect(getByTestId("search-results__image")).toBeVisible();
  });
});
