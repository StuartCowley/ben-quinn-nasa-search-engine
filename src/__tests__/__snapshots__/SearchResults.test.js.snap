// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <p>
    Search results:
  </p>
  <img
    alt="card"
    class="search-results__image"
    data-testid="search-results__image"
    src="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
  />
</DocumentFragment>
`;
